language/stdlib/transaction_scripts/unfreeze_account.move

```diff
@@ -1,11 +1,16 @@
 script {
 use 0x1::LibraAccount;
 use 0x1::SlidingNonce;
+use 0x1::Roles::{Self, AccountUnfreezing};
 
 /// Script for un-freezing account by authorized initiator
 /// sliding_nonce is a unique nonce for operation, see sliding_nonce.move for details
 fun main(account: &signer, sliding_nonce: u64, to_unfreeze_account: address) {
     SlidingNonce::record_nonce_or_abort(account, sliding_nonce);
-    LibraAccount::unfreeze_account(account,  to_unfreeze_account);
+    let role = LibraAccount::extract_role(account);
```

Comment:
After reading through a few of these, I'm wondering if:
- `RoleID` wants to be a top-level resource instead of inside `LibraAccount`
- Instead of extracting/restoring the `RoleID`, `Roles::extract_privileged_to_capability` can take a `&signer`, look at the `RoleID` under `&signer`, and give back the relevant privilege

This design is possible because you have a discipline of always passing around `Capability`'s (never `RoleId`s), which is interesting. Do you think we would ever want to pass a `&RoleID` to something other than account creation?

Reply:
I really like this idea, and it simplifies a lot of things. The other thing is that this also means that `&RoleId` will never be passed as a parameter which makes me very happy... 

This also makes it provable that a role can't change (there's one `move_to`, and no `move_from` from there so the `RoleId` won't need to hold the `owner_address`!)